### ISSUE TITLE
oAuth client disabled

### ISSUE THREAD

**Issue by Foovanadil** [Microsoft/google-play-vsts-extension#99]
Not sure if this is an extension issue or a change on google's side or what.

All was working fine until 10 days ago (which was my last attempted build).

Now when going to publish to to the Play store I get an error saying the oAuth client is disabled.

I am using a Service Endpoint. I tried switching to JSON auth file and received the same error. (the log below is from my attempt with a JSON file, NOTE: I stripped my apk file name out for privacy reasons and replaced it with MY APK)

I tried disconnecting the Service endpoint and re-creating it, still the same error. The credentials I am using are correct (haven't changed) and the user being used is not expired in the play console (set to never expire).

Any ideas?

```
Task         : Google Play - Release
Description  : Release an app to the Google Play Store
Version      : 2.135.0
Author       : Microsoft Corporation
Help         : 
==============================================================================
agent.TempDirectory=D:\a\_temp
loading inputs and endpoints
loading ENDPOINT_AUTH_5931cf8c-fd1c-446f-9bae-6a41fbcfa7e9
loading ENDPOINT_AUTH_PARAMETER_5931cf8c-fd1c-446f-9bae-6a41fbcfa7e9_PASSWORD
loading ENDPOINT_AUTH_PARAMETER_5931cf8c-fd1c-446f-9bae-6a41fbcfa7e9_USERNAME
loading ENDPOINT_AUTH_PARAMETER_SYSTEMVSSCONNECTION_ACCESSTOKEN
loading ENDPOINT_AUTH_SCHEME_5931cf8c-fd1c-446f-9bae-6a41fbcfa7e9
loading ENDPOINT_AUTH_SCHEME_SYSTEMVSSCONNECTION
loading ENDPOINT_AUTH_SYSTEMVSSCONNECTION
loading INPUT_APKFILE
loading INPUT_AUTHTYPE
loading INPUT_CHANGELOGFILE
loading INPUT_LANGUAGECODE
loading INPUT_METADATAROOTPATH
loading INPUT_SERVICEACCOUNTKEY
loading INPUT_SERVICEENDPOINT
loading INPUT_SHOULDATTACHMETADATA
loading INPUT_SHOULDUPLOADAPKS
loading INPUT_TRACK
loading INPUT_VERSIONCODEFILTERTYPE
loading SECRET_KEYSTOREPASSWORD
loaded 19
Agent.ProxyUrl=undefined
Agent.CAInfo=undefined
Agent.ClientCert=undefined
check path : D:\a\_tasks\GooglePlayRelease_8cf7cac0-620b-11e5-b4cf-8565e60f4d27\2.135.0\task.json
set resource file to: D:\a\_tasks\GooglePlayRelease_8cf7cac0-620b-11e5-b4cf-8565e60f4d27\2.135.0\task.json
system.culture=en-US
Prepare task inputs.
authType=JsonFile
serviceAccountKey=D:\a\1\s\Certs\googleplaykey.json
check path : D:\a\1\s\Certs\googleplaykey.json
apkFile=MY APK
Main APK pattern:MY APK
check path : MY APK
Found main APK to upload: MY APK (version code 215)
Found the main APK file: MY APK (version code 215).
additionalApks=null
versionCodeFilterType=all
track=alpha
userFraction=null
shouldAttachMetadata=false
shouldUploadApks=true
changelogFile=D:\a\1\s
languageCode=en-US
Getting a package name from MY APK
Updating Global Parameter packageName to "MY PACAKGE"
... updated.
Initializing JWT.
Initializing Google Play publisher API.
Authorize JWT.
task result: Failed
Error: disabled_client: The OAuth client was disabled.
Processed: ##vso[task.issue type=error;]Error: disabled_client: The OAuth client was disabled.
Processed: ##vso[task.complete result=Failed;]Error: disabled_client: The OAuth client was disabled.
```

**Comment by madhurig** [comment]
Hi,

Are you able to upload the APK from the Google Developer Console? Based on the error it seems like your App has been blocked by Google. See related thread here: https://stackoverflow.com/questions/24686567/google-oauth-authentication-suddenly-fails-and-issues-disabled-client 

Thanks,
Madhuri

**Comment by Foovanadil** [comment]
Hmm, I don't see any indication that app has been banned. I am able to manage releases and create new releases in the console. Maybe just the oAuth piece has been disabled? Any idea how to check that?

**Comment by madhurig** [comment]
The error is being returned when the task tries to authenticate with google. We use the email and private key from the JSON obtained from the Google developer console when you create a service account and grant it permissions. Can you re-check the permissions for the service account in your google developer console? Try regenerating the keys and updating them in the service endpoint used in the task. 

You can re-check the pre-requisites section to verify if any of your credentials used for the CD process have expired: https://github.com/Microsoft/google-play-vsts-extension#prerequisites 

Since you are able to otherwise update your app, I would suspect the issue with those credentials but the server is providing a confusing error message. 

Thanks,
Madhuri

**Comment by Foovanadil** [comment]
Thanks for the quick reply. I am trying to re-create the service account to see if that fixes it.

In the meantime I just realized this might not be the right place for this feedback. I am using the VSTS built in Google Play step (visualstudio.com) 

Is this the same extension that runs there?

**Comment by madhurig** [comment]
@Foovanadil : This is the right location. The GooglePlay task in VSTS is provided by this extension :). Let us know if you have any success after recreating the service account.

**Comment by lkillgore** [comment]
Hi @Foovanadil, I'm closing this bug due to inactivity.  If this is still an issue, please let us know and open a new ticket.  Thanks!
